lib/resty/mlcache.lua

```diff
@@ -412,15 +407,30 @@ function _M:get(key, opts, cb, ...)
         return unlock_and_ret(lock, data, nil, 2)
     end
 
+    -- opts validation
+
+    local ttl, neg_ttl = check_opts(self, opts)
```

Comment:
Hmmm I think this should move to L. 384. We probably don't want to run the L3 callback if we already know the options are not valid, since it would be a wasted I/O cycle. But since L. 384 is a cold code path (it only runs when both the L1 and the L2 caches are missed), I think it is a reasonable compromise?